src/main/java/org/opendevstack/provision/services/ConfluenceAdapter.java

```diff
@@ -201,40 +201,37 @@ int updateSpacePermissions (ProjectData data, String crowdCookieValue) throws IO
       int updatedPermissions = 0;
 
       logger.debug("Found permissionsets: "+ permissionFiles.length);
-      
-      for (int i = 0; i < permissionFiles.length; i++)
-      {
-    	  String permissionFilename = permissionFiles[i].getFilename();
-    	  
-    	  BufferedReader reader = new BufferedReader (new InputStreamReader(permissionFiles[i].getInputStream()));
-    	  String permissionset = null;
-    	  try 
-    	  {
-    		  // we know it's a singular pseudo json line
-    		  permissionset = reader.readLine();
-    	  } finally 
-    	  {
-    		  //sq finding
-    		  reader.close();
-    	  }
-    	  
-    	  permissionset = permissionset.replace("SPACE_NAME", data.key);
-    	  
-    	  if (permissionFilename.contains("adminGroup")) {
-    		  permissionset = permissionset.replace(SPACE_GROUP, data.adminGroup);
-    	  } else if (permissionFilename.contains("userGroup")) {
-    		  permissionset = permissionset.replace(SPACE_GROUP, data.userGroup);
-      	  } else if (permissionFilename.contains("readonlyGroup")) {
-    		  permissionset = permissionset.replace(SPACE_GROUP, data.readonlyGroup);  
-      	  } else if (permissionFilename.contains("keyuserGroup")) {
-    		  permissionset = permissionset.replace(SPACE_GROUP, globalKeyuserRoleName);  
-      	  } 
-    	  
-    	  String path = String.format("%s%s/addPermissionsToSpace", confluenceUri, confluenceLegacyApiPath);
-    	  
-    	  client.callHttp(path, permissionset, crowdCookieValue, false, RestClient.HTTP_VERB.POST, String.class);
-    	  
-	      updatedPermissions++;
+
+      for (Resource permissionFile : permissionFiles) {
+          String permissionFilename = permissionFile.getFilename();
+
+          BufferedReader reader = new BufferedReader(new InputStreamReader(permissionFile.getInputStream()));
+          String permissionset = null;
+          try {
+              // we know it's a singular pseudo json line
+              permissionset = reader.readLine();
+          } finally {
+              //sq finding
+              reader.close();
+          }
+
```

Comment:
should be "try with resources" as well - like in [Introduce automatic resource management](https://github.com/opendevstack/ods-provisioning-app/pull/124/commits/5c05b2ce951df87b6ab492af02cafdb756e7c4a3)